aplib.net-demo/Assets/Scripts/WFC/GridPlacer.cs

```diff
@@ -228,30 +239,104 @@ private void PlaceDoors(int x, int z, Room room)
+                    Teleporter.Teleporter entryTeleporter = PlaceTeleporter(CentreOfCell(nextEntryCell) + teleporterHeightOffset);
+
+                    // Link the entry teleporter back to the previous connected component, bidirectionally.
+                    previousExitTeleporter!.TargetTeleporter = entryTeleporter;
+                    entryTeleporter.TargetTeleporter = previousExitTeleporter;
+
+                    // Create an OffMeshLink between the two teleporters.
+                    OffMeshLink offMeshLink = previousExitTeleporter.gameObject.AddComponent<OffMeshLink>();
+                    offMeshLink.startTransform = previousExitTeleporter.transform;
+                    offMeshLink.endTransform = entryTeleporter.transform;
+                    offMeshLink.autoUpdatePositions = true;
+                    offMeshLink.area = NavMesh.GetAreaFromName("PlayerWalkable");
+                }
+
+                // Place the exit teleporter of the current connected component at the end of the connected component.
+                // (Assuming that the connected component has at least two cells, `nextExitCell` will never equal `nextEntryCell`)
+                Cell nextExitCell = connectedComponentsEnumerator.Current!.Last();
+                Teleporter.Teleporter exitTeleporter = PlaceTeleporter(CentreOfCell(nextExitCell) + teleporterHeightOffset);
+
+                // Update iteration progress
+                previousExitTeleporter = exitTeleporter;
+                connectedComponentsProcessed++;
             }
 
-            // TODO the joining of the connected components with teleporters will be done in another PR.
+            // If at least one exit portal has been placed, the final exit teleporter will not be linked to
+            // another connected component, for this final exit teleporter belongs to the final connected component.
+            if (connectedComponentsProcessed > 0)
+            {
+                Destroy(previousExitTeleporter!.gameObject);
+            }
         }
 
-        // Here are temporary helper methods used to display the connected components in different colors.
-        private static readonly Color[] _colors =
+        /// <summary>
+        /// Places a teleporter at a given location.
+        /// </summary>
+        /// <param name="coordinates">The coordinates of the teleporter.</param>
+        /// <returns>A reference to the <see cref="Teleporter"/> component of the teleporter.</returns>
+        /// <remarks>All teleporters are clustered under a 'Teleporter' empty gameobject.</remarks>
+        private Teleporter.Teleporter PlaceTeleporter(Vector3 coordinates)
         {
-            Color.red, Color.blue, Color.green, Color.yellow, Color.magenta, Color.cyan
-        };
+            // Give all teleporters a parent object for organization
+            GameObject teleportersParent = GameObject.Find("Teleporters") ?? new GameObject("Teleporters");
 
```

Comment:
I know this works, for the scene initially does not even have a `Telepoters` GameObject.

Reply:
It works, but Unity does not like it, and also gives me warnings, as Unity is very cursed it is probably a good idea to abide by its warnings, or it will calculate the perfect time to fail us so we in turn fail the software project.

Reply:
I would be curious how those warnings are explained, but I will change it.